kafka-handler/src/test/org/apache/hadoop/hive/kafka/KafkaDagCredentialSupplierTest.java

```diff
@@ -75,13 +93,39 @@ public static void stopCluster() {
   }
 
   @Test
-  public void testObtainTokenNotNull() {
+  public void testObtainTokenFromSamlPlainTextListenerNotNull() {
+    Properties props = new Properties();
+    props.setProperty("kafka.bootstrap.servers", KafkaBrokerResource.BROKER_SASL_PORT);
+    checkObtainToken(props);
+  }
+
+  @Test
+  public void testObtainTokenFromSamlSslListenerNotNull()
+      throws IOException, URISyntaxException, KeyStoreException, CertificateException, NoSuchAlgorithmException {
+    Properties props = new Properties();
+    props.setProperty(KafkaTableProperties.HIVE_KAFKA_BOOTSTRAP_SERVERS.getName(),
+        KafkaBrokerResource.BROKER_SASL_SSL_PORT);
+    // Should the SSL properties be divided between producer/consumer? Probably not!
+    props.setProperty(SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SASL_SSL.name);
+    props.setProperty(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, "");
+    String pwdAlias = "HereIsAnAliasForTheKeyWhichHoldsTheTrustorePassword";
+    URI storeURI = createCredentialStore(ImmutableMap.of(pwdAlias, KAFKA_BROKER_RESOURCE.getTruststorePwd()));
+    props.setProperty(KafkaTableProperties.HIVE_KAFKA_SSL_CREDENTIAL_KEYSTORE.getName(), storeURI.toString());
+    props.setProperty(KafkaTableProperties.HIVE_KAFKA_SSL_TRUSTSTORE_PASSWORD.getName(), pwdAlias);
+    props.setProperty(KafkaTableProperties.HIVE_KAFKA_SSL_KEYSTORE_PASSWORD.getName(), "");
+    props.setProperty(KafkaTableProperties.HIVE_KAFKA_SSL_KEY_PASSWORD.getName(), "");
+    String location = KAFKA_BROKER_RESOURCE.getTruststorePath().toUri().toString();
```

Comment:
``getTruststorePath()`` by code can return ``null`` as well, so, if does we would fail with NPE here
```
  Path getTruststorePath() {
    return truststoreFile != null ? truststoreFile.toPath() : null;
  }
```